### ISSUE TITLE
Revert changes to Follow button and restore the "follows you" display indicator

### ISSUE THREAD

**Issue by brookmiles** [mastodon/mastodon#32604]
### Pitch

Revert https://github.com/mastodon/mastodon/pull/28452 and related UI changes in v4.3.0 to the "Follow/Unfollow" button, and restore the "follows you" display indicator.

### Motivation

The new behaviour of the Follow/Unfollow/Follow back/Mutual button is confusing, less clear than the previous design, and violates standard UI conventions.

As a UI element, a button should always perform an action, and it should be labelled with the action that it will perform.  When the button is labelled "Follow", pressing the button will perform a follow action, when the button is labelled "Unfollow", the button will perform an unfollow action.

The button now displays "Mutual" when you follow the user being viewed, and they also follow you, however pressing the button does not perform a "Mutual" action.  Not only is "Mutual" not an action or verb, but the action that actually takes place is "Unfollow". The button should therefor be labelled "Unfollow" in this situation.

![new design Unfollow button which is labelled Mutual](https://github.com/user-attachments/assets/d0b75154-d8b8-43d0-a938-75141083d538)

Previously, there was a clear indicator on the profile page that the user being viewed was following you.  This is now indicated by the changes in wording of the "Follow" button.  This is confusing in several ways 
- a basic button should not be a primary indicator of state, and it is not the obvious place to find this information
- the button performs actions controlling whether you follow the profile being viewed, but the button displays state based on whether the profile being viewed follows you
- the word "Mutual" displayed on a button (an issue by itself, as mentioned above), is less obviously related to whether the user being viewed in following you than the previously displayed "Follows you" indicator, which was clearly an indicator that the profile follows you and was not an action that you perform as a user.

**Previous design:**

The button performing the "unfollow" action is clearly labelled "Unfollow", and the fact that the profile is following you is clearly indicated by the presence of the "Follows you" tag.

![Profile viewed with previous design, a 'Follows you' tag, and the Unfollow button is labelled Unfollow](https://github.com/user-attachments/assets/c2dcb971-e9b5-445b-b73a-991902ad0852)

**Comment by ThisIsMissEm** [comment]
Feels like maybe the follows you / mutual thing could be a badge next to the domain information?

**Comment by alexcat3** [comment]
@ThisIsMissEm Yeah I think that the original follows you tag was a bit easy to overlook, but having a button labeled with something that is not a verb is a bad idea. Your idea of placing it next to the domain would make it more noticeable while avoiding the current confusing interface.

**Comment by smiba** [comment]
Agree with this, I don't think this is good design and the old "Unfollow" should return

**Comment by JamesLavin** [comment]
I was so confused by and unsure about that button that I started Googling and landed on this page.
